### ISSUE TITLE
✨ Dialogs for applicant dashboard

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#12214]
## ✨ Feature

Create dialogs for use on the applicant dashboard page with the Applications and Processes widget #10788.

**OUT OF SCOPE**
- Multiple departments for application dialogs
- Flexible work location options in dialogs
- Making Hired status work with Search depending on the suspendedAt date - see #12212

## 🕵️ Details

The Applications and Processes widget should be getting done in #10788. These dialogs, which appear if you click on the Applications and Processes cards, can be worked on in parallel.

Both sets of dialogs (application and process) follow the same pattern as the dialog for requests that was created in #11653 for the manager dashboard.

The statuses displayed inside the dialogs are described (and copy provided for the descriptions you need for this issue) in #12164. And that issue will actually need to be completed for this one since the existing utility function for statuses doesn't provide or line up with the descriptions.

### Application tool dialogs
- Each dialog shows a status summary at the top.
- Below status is a section with basic info
   - Job title, classification, salary range, work stream &  language requirement
   - Department should be singular for now, ie. not a list (we can switch it to a list when we allow for multiple departments on the job ad)
- Then we show dates (still in the same section)
   - For Draft: Show **deadline to apply**
   - For Draft (when less than 3 days remain to apply), and Expired: Show **deadline to apply** in Red.
   - For all others: Show **Received** which is the `submittedAt` date. (reminder to display in timezone of the user)
- Next section shows more details about the process
   - Employment length, Work location, Security clearance, process number
   - **Flexible work location** options is **out of scope** for this ticket
- In the next section, we list the skills from the process in accordions
   - Inside the accordions are just plain text lists of the skills (image below in screenshots section)
- Final section is links
   - All dialogs have a Cancel link (closes dialog)
   - All dialogs have a View job advertisement link
   - Draft job applications have "Continue application" button/link
   - All dialogs other than Draft have a "View application" button/link
   - For Successful, the "View application" is just a simple link
   - For Successful, there is a "View recruitment process" button/link that **Closes the dialog**, **Expands the processes tool**, and **focuses the related process**

### Processes tool dialogs
**Same as applications except**
- Statuses that link to "at the end of this dialog" should link to the "Your availability" select input.
- Statuses that link to "reach out to the functional community" should link to Support page
- The dates we show:
  - Awarded is the "finalDecisionAt" date.
  - Expires is the expiry date of the candidate in the pool/process
- Congratulations & manage suspension section
   - This section shows up for all process statuses **Except Expired**
   - Saving "I'm not currently interested..." sets the "suspendedAt" date to the current dateTime (likely changing the status for this process)
   - Saving the "I'm available for hire and interested..." clears (nulls) the "suspendedAt" date (likely changing the status for this process)
   - Actually getting "suspendedAt" working with the Hired status is out of scope for this issue (see #12212)
-  Final section links
   - View application and View job ad links always shown
   - Save changes button/link and cancel link shown when suspension section is shown
   - Close button/link shown for Expired and Removed statuses

## 🧑‍🎨 Designer

@JoshBeveridge 

https://www.figma.com/design/oZhQe9ahidHFfaiqgREvxa/Dashboard-(All-users)?node-id=2011-38235&node-type=frame&t=sltU21dpZZlO5BUt-0

## 📸 Screenshot

Example of skills expanded in application dialog
![image](https://github.com/user-attachments/assets/8a913282-3843-49f1-b9fd-f43fbb2abbf4)

## 🙋‍♀️ Proposed Implementation

Start from what we built for Manager Dashboard.
The dialogs for Applications and Processes do not necessarily need to be separate things in the code. They follow a similar pattern, but up to the devs to decide.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Applications dialog displays correctly
- [ ] Processes dialog displays correctly

## 🛑 Blockers

```[tasklist]
- [ ] #12164
```

**Comment by esizer** [comment]
> For Successful, there is a "View recruitment process" button/link that Closes the dialogue, Expands the processes tool, and focuses the related process

Do we have a "processes tool" yet? 

> Close button/link shown for Expired and Removed statuses

What would this button do. Do we have the ability to close an application currently?
